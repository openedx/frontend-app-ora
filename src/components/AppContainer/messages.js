import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  ERR_INVALID_STATE_FOR_ASSESSMENT: {
    defaultMessage: 'This step is not available. Unable to retrieve the assessment.',
    description: 'Error message for invalid state for assessment',
    id: 'frontend-app-ora.StatusAlerts.ERR_INVALID_STATE_FOR_ASSESSMENT',
  },
  unknownError: {
    defaultMessage: 'An unknown error occurred. Please try again.',
    description: 'Error message for unknown error',
    id: 'frontend-app-ora.StatusAlerts.unknownError',
  },
  errorHeader: {
    defaultMessage: 'Something went wrong.',
    description: 'Error message header',
    id: 'frontend-app-ora.StatusAlerts.errorHeader',
  },
});

export default messages;
