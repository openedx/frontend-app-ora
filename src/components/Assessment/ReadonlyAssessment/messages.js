import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  readMore: {
    id: 'frontend-app-ora.readMore',
    defaultMessage: 'Read more',
    description: 'Read more button text',
  },
  readLess: {
    id: 'frontend-app-ora.readLess',
    defaultMessage: 'Read less',
    description: 'Read less button text',
  },
  unweightedGrade: {
    id: 'ora-collapsible-comment.unweightedGrade',
    defaultMessage: '{stepLabel} grade',
    description: 'Unweighted grade group text',
  },
  grade: {
    id: 'frontend-app-ora.grade',
    defaultMessage: '{stepLabel} grade:',
    description: 'Grade button text',
  },
  gradePoints: {
    id: 'frontend-app-ora.gradePoints',
    defaultMessage: '{earned} / {possible}',
    description: 'Grade points button text',
  },
  notWeightedGradeLabel: {
    id: 'frontend-app-ora.notWeightedGradeLabel',
    defaultMessage: '(Not weighted toward final grade))',
    description: 'Not weighted grade label',
  },
  overallFeedback: {
    id: 'frontend-app-ora.overallFeedback',
    defaultMessage: 'Overall feedback',
    description: 'Overall feedback label',
  },
  stepComments: {
    id: 'ora-collapsible-comment.stepComments',
    defaultMessage: '{step} comments',
    description: 'Step comments label for completed assessments',
  },
  comments: {
    id: 'ora-collapsible-comment.comments',
    defaultMessage: 'comments',
    description: 'Comments label for completed assessments',
  },
  submittedGrade: {
    id: 'ora-collapsible-comment.submittedGrade',
    defaultMessage: 'Submitted grade',
    description: 'Header for submitted grade display',
  },
  submittedAssessment: {
    id: 'ora-collapsible-comment.submittedAssessment',
    defaultMessage: 'Submitted assessment',
    description: 'Header for submitted assessment display',
  },
  comment: {
    id: 'ora-collapsible-comment.comment',
    defaultMessage: 'Comments',
    description: 'Header for comment component if no step provided',
  },
  stepComment: {
    id: 'ora-collapsible-comment.stepComment',
    defaultMessage: '{step} comment',
    description: 'Header for comment component if step is provided',
  },
  points: {
    id: 'ora-collapsible-comment.points',
    defaultMessage: 'Points',
    description: 'Header for points component',
  },
});

export default messages;
