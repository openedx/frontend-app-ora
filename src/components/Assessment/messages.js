import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  gradeSubmitted: {
    id: 'frontend-app-ora.EditableAssessment.gradeSubmitted',
    defaultMessage: 'Grade Submitted',
    description: 'Submit Grade button text after successful submission',
  },
  rubric: {
    id: 'frontend-app-ora.EditableAssessment.rubric',
    defaultMessage: 'Rubric',
    description: 'Rubric interface label',
  },
  submitGrade: {
    id: 'frontend-app-ora.EditableAssessment.submitGrade',
    defaultMessage: 'Submit grade',
    description: 'Submit Grade button text',
  },
  submittingGrade: {
    id: 'frontend-app-ora.EditableAssessment.submittingGrade',
    defaultMessage: 'Submitting grade',
    description: 'Submit Grade button text while submitting',
  },
  overallComments: {
    id: 'frontend-app-ora.EditableAssessment.overallComments',
    defaultMessage: 'Overall comments',
    description: 'Rubric overall comments label',
  },
  addComments: {
    id: 'frontend-app-ora.EditableAssessment.addComments',
    defaultMessage: 'Add comments (Optional)',
    description: 'Rubric comments input label',
  },
  comments: {
    id: 'frontend-app-ora.EditableAssessment.comments',
    defaultMessage: 'Comments (Optional)',
    description: 'Rubric comments display label',
  },
  overallFeedbackError: {
    id: 'frontend-app-ora.EditableAssessment.overallFeedbackError',
    defaultMessage: 'The overall feedback is required',
    description: 'Error message when feedback input is required',
  },
  finishLater: {
    id: 'frontend-app-ora.EditableAssessment.finishLater',
    defaultMessage: 'Finish grading later',
    description: 'Button text for close modal action in the assessment view',
  },
});

export default messages;
