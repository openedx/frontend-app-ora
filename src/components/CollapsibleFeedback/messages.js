import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  readMore: {
    id: 'ora-collapsible-comment.readMore',
    defaultMessage: 'Read more',
    description: 'Read more button text',
  },
  readLess: {
    id: 'ora-collapsible-comment.readLess',
    defaultMessage: 'Read less',
    description: 'Read less button text',
  },
  grade: {
    id: 'ora-collapsible-comment.grade',
    defaultMessage: '{stepLabel} Grade:',
    description: 'Grade button text',
  },
  gradePoints: {
    id: 'ora-collapsible-comment.gradePoints',
    defaultMessage: '{earned} / {possible}',
    description: 'Grade points button text',
  },
  notWeightedGradeLabel: {
    id: 'ora-collapsible-comment.notWeightedGradeLabel',
    defaultMessage: '(Not weighted toward final grade))',
    description: 'Not weighted grade label',
  },
  overallFeedback: {
    id: 'ora-collapsible-comment.overallFeedback',
    defaultMessage: 'Overall Feedback',
    description: 'Overall feedback label',
  },
});

export default messages;
