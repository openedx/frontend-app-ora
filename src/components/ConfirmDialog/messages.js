import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  goBack: {
    defaultMessage: 'Go back',
    description: 'Label for the cancel button',
    id: 'frontend-app-ora.ConfirmDialog.goBack',
  },
});

export default messages;
