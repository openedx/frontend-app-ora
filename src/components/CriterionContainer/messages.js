import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  addComments: {
    id: 'frontend-app-ora.CriterionFeedback.addCommentsLabel',
    defaultMessage: 'Add comments',
    description: 'label for editable feedback field',
  },
  comments: {
    id: 'frontend-app-ora.CriterionFeedback.commentsLabel',
    defaultMessage: 'Comments',
    description: 'label for read-only feedback field',
  },
  optional: {
    id: 'frontend-app-ora.CriterionFeedback.optional',
    defaultMessage: '(Optional)',
    description: 'additional label for optional feedback field',
  },
  optionPoints: {
    id: 'frontend-app-ora.RadioCriterion.optionPoints',
    defaultMessage: '{points} points',
    description: 'criterion option point value display',
  },
  rubricSelectedError: {
    id: 'frontend-app-ora.RadioCriterion.rubricSelectedError',
    defaultMessage: 'Rubric selection is required',
    description: 'Error message when rubric radio did not get selected',
  },
  criterionFeedbackError: {
    id: 'frontend-app-ora.CriterionFeedback.criterionFeedbackError',
    defaultMessage: 'The feedback is required',
    description: 'Error message when feedback is required',
  },
});

export const trainingMessages = defineMessages({
  valid: {
    id: 'frontend-app-ora.TrainingCriterion.valid',
    defaultMessage: 'Good job!',
    description: 'Training criterion valid',
  },
  invalid: {
    id: 'frontend-app-ora.TrainingCriterion.invalid',
    defaultMessage: 'Reevaluate and select a new score',
    description: 'Error message when select wrong student training',
  },
});

export default messages;
