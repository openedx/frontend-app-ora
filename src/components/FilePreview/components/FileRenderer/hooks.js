import { useState } from 'react';
import { errorStatuses, errorMessages, renderers } from '../constants';
import { getFileType } from '../utils';
import messages from './messages';

/**
 * component hooks
 */
export const useRenderData = ({
  file,
  formatMessage,
}) => {
  const [errorStatus, setErrorStatus] = useState(null);
  const [isLoading, setIsLoading] = useState(true);

  const setState = (newState) => {
    setErrorStatus(newState.errorStatus);
    setIsLoading(newState.isLoading);
  };

  const stopLoading = (status = null) => setState({ isLoading: false, errorStatus: status });

  const errorMessage = (
    errorMessages[errorStatus] || errorMessages[errorStatuses.serverError]
  );
  const errorAction = {
    id: 'retry',
    onClick: () => setState({ errorStatus: null, isLoading: true }),
    message: messages.retryButton,
  };
  const error = {
    headerMessage: errorMessage,
    children: formatMessage(errorMessage),
    actions: [errorAction],
  };

  const Renderer = renderers[getFileType(file.fileName)];
  const rendererProps = {
    fileName: file.fileName,
    url: file.fileUrl,
    onError: stopLoading,
    onSuccess: () => stopLoading(),
  };

  return {
    errorStatus,
    isLoading,
    error,
    Renderer,
    rendererProps,
  };
};
