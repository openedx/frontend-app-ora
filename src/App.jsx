import React from 'react';
import { Routes, Route } from 'react-router-dom';
import { ErrorPage } from '@edx/frontend-platform/react';
import { useIntl } from '@edx/frontend-platform/i18n';

import AssessmentView from 'views/AssessmentView';
import SubmissionView from 'views/SubmissionView';
import XBlockView from 'views/XBlockView';
import XBlockStudioView from 'views/XBlockStudioView';
import GradeView from 'views/GradeView';

import PageRoute from 'components/PageRoute';

import { useHandleModalCloseEvent } from 'hooks/modal';

import messages from './messages';
import routes from './routes';

const App = () => {
  const { formatMessage } = useIntl();
  const handleModalClose = useHandleModalCloseEvent();

  React.useEffect(() => {
    window.addEventListener('message', handleModalClose);
    return () => window.removeEventListener('message', handleModalClose);
  }, [handleModalClose]);

  return (
    <Routes>
      <Route
        path={routes.xblock}
        element={(<PageRoute><XBlockView /></PageRoute>)}
      />
      <Route
        path={routes.xblockStudio}
        element={(<PageRoute><XBlockStudioView /></PageRoute>)}
      />
      <Route
        path={routes.xblockPreview}
        element={(<PageRoute><XBlockView /></PageRoute>)}
      />
      <Route
        path={routes.peerAssessment}
        element={(<PageRoute isModal><AssessmentView /></PageRoute>)}
      />
      <Route
        path={routes.selfAssessment}
        element={(<PageRoute isModal><AssessmentView /></PageRoute>)}
      />
      <Route
        path={routes.studentTraining}
        element={(<PageRoute isModal><AssessmentView /></PageRoute>)}
      />
      <Route
        path={routes.submission}
        element={(<PageRoute isModal><SubmissionView /></PageRoute>)}
      />
      <Route
        path={routes.graded}
        element={(<PageRoute isModal><GradeView /></PageRoute>)}
      />
      <Route
        key="error"
        path={routes.root}
        element={<ErrorPage message={formatMessage(messages.error404Message)} />}
      />
    </Routes>
  );
};

export default App;
