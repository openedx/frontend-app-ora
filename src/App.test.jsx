import React from 'react';
import { when } from 'jest-when';
import { Route, Routes } from 'react-router-dom';

import { ErrorPage } from '@edx/frontend-platform/react';
import { useIntl } from '@edx/frontend-platform/i18n';
import { formatMessage, shallow } from '@edx/react-unit-test-utils';

import AssessmentView from 'views/AssessmentView';
import SubmissionView from 'views/SubmissionView';
import XBlockView from 'views/XBlockView';
import XBlockStudioView from 'views/XBlockStudioView';
import GradeView from 'views/GradeView';

import PageRoute from 'components/PageRoute';

import { useHandleModalCloseEvent } from 'hooks/modal';

import messages from './messages';
import routes from './routes';

import App from './App';

jest.mock('react-router-dom', () => ({
  Routes: 'Routes',
  Route: 'Route',
}));

jest.mock('@edx/frontend-platform/react', () => ({
  AuthenticatedPageRoute: 'AuthenticatedPageRoute',
  ErrorPage: 'ErrorPage',
}));
jest.mock('views/AssessmentView', () => 'AssessmentView');
jest.mock('views/SubmissionView', () => 'SubmissionView');
jest.mock('views/XBlockView', () => 'XBlockView');
jest.mock('views/XBlockStudioView', () => 'XBlockStudioView');
jest.mock('views/GradeView', () => 'GradeView');
jest.mock('components/PageRoute', () => 'PageRoute');

jest.mock('hooks/modal', () => ({
  useHandleModalCloseEvent: jest.fn(),
}));

const handleModalClose = jest.fn();
when(useHandleModalCloseEvent).calledWith().mockReturnValue(handleModalClose);
const addEventListener = jest.fn();
const removeEventListener = jest.fn();

let el;
describe('App component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(window, 'addEventListener').mockImplementation(addEventListener);
    jest.spyOn(window, 'removeEventListener').mockImplementation(removeEventListener);
    el = shallow(<App />);
  });
  describe('behavior', () => {
    it('initializes i18n and refresh event from hooks', () => {
      expect(useIntl).toHaveBeenCalled();
      expect(useHandleModalCloseEvent).toHaveBeenCalled();
    });
    it('adds handler for modal close event that refreshes page data', () => {
      expect(React.useEffect.mock.calls.length).toEqual(1);
      const [[effect, prereqs]] = React.useEffect.mock.calls;
      expect(prereqs).toEqual([handleModalClose]);
      const out = effect();
      expect(addEventListener).toHaveBeenCalledWith('message', handleModalClose);
      out();
      expect(removeEventListener).toHaveBeenCalledWith('message', handleModalClose);
    });
  });
  describe('render', () => {
    test('snapshot', () => {
      expect(el.snapshot).toMatchSnapshot();
    });
    const testComponent = (toTest, { route, Component, isModal }) => {
      expect(toTest.type).toEqual(Route);
      expect(toTest.props.path).toEqual(route);
      const { element } = toTest.props;
      expect(toTest.props.element.type).toEqual(PageRoute);
      if (isModal) {
        expect(toTest.props.element.props.isModal).toEqual(true);
      }
      const expectedElement = shallow(<PageRoute><Component /></PageRoute>);
      expect(shallow(element)).toMatchObject(expectedElement);
    };
    const testAssessmentRoute = (toTest, { route }) => {
      testComponent(toTest, { route, Component: AssessmentView, isModal: true });
    };
    test('route order', () => {
      const renderedRoutes = el.instance.findByType(Routes)[0].children;
      testComponent(renderedRoutes[0], { route: routes.xblock, Component: XBlockView });
      testComponent(renderedRoutes[1], { route: routes.xblockStudio, Component: XBlockStudioView });
      testComponent(renderedRoutes[2], { route: routes.xblockPreview, Component: XBlockView });
      testAssessmentRoute(renderedRoutes[3], { route: routes.peerAssessment });
      testAssessmentRoute(renderedRoutes[4], { route: routes.selfAssessment });
      testAssessmentRoute(renderedRoutes[5], { route: routes.studentTraining });
      testComponent(renderedRoutes[6], {
        route: routes.submission,
        Component: SubmissionView,
        isModal: true,
      });
      testComponent(renderedRoutes[7], {
        route: routes.graded,
        Component: GradeView,
        isModal: true,
      });
      expect(renderedRoutes[8].matches(shallow(
        <Route
          key="error"
          path={routes.root}
          element={<ErrorPage message={formatMessage(messages.error404Message)} />}
        />,
      )));
    });
  });
});
