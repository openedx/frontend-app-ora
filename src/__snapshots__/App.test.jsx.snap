// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render snapshot 1`] = `
<Routes>
  <Route
    element={
      <PageRoute>
        <XBlockView />
      </PageRoute>
    }
    path="xblock/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute>
        <XBlockStudioView />
      </PageRoute>
    }
    path="xblock_studio/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute>
        <XBlockView />
      </PageRoute>
    }
    path="xblock_preview/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute
        isModal={true}
      >
        <AssessmentView />
      </PageRoute>
    }
    path="peer_assessment/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute
        isModal={true}
      >
        <AssessmentView />
      </PageRoute>
    }
    path="self_assessment/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute
        isModal={true}
      >
        <AssessmentView />
      </PageRoute>
    }
    path="student_training/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute
        isModal={true}
      >
        <SubmissionView />
      </PageRoute>
    }
    path="submission/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <PageRoute
        isModal={true}
      >
        <GradeView />
      </PageRoute>
    }
    path="graded/:courseId/:xblockId/:progressKey?"
  />
  <Route
    element={
      <ErrorPage
        message="Page not found"
      />
    }
    key="error"
    path="/*"
  />
</Routes>
`;
